desk/app/channels.hoon

```diff
@@ -730,75 +898,63 @@
   =?  +.pole  !?=([?(%v0 %v1 %v2 %v3 %v4) *] +.pole)
     [%v0 +.pole]
   ?+    pole  [~ ~]
-      [%x ?(%v0 %v1) %channels ~]   ``channels+!>((uv-channels-1:utils v-channels))
     ::
-        [%x %v2 %channels full=?(~ [%full ~])]
-      ``channels-2+!>((uv-channels-2:utils v-channels ?=(^ full.pole)))
+    ::    /x/v/channels: get unversioned channels
     ::
-      [%x ?(%v0 %v1) %init ~]    ``noun+!>([unreads (uv-channels-1:utils v-channels)])
-      [%x %v2 %init ~]  ``noun+!>([unreads (uv-channels-2:utils v-channels |)])
+      [%x ?(%v0 %v1) %channels ~]
+    ``channels+!>(`channels-0:c`(uv-channels-1:utils v-channels))
     ::
-        [%x %v3 %init ~]
-      =/  init  [(uv-channels-2:utils v-channels |) hidden-posts]
-      ``noun+!>(`[channels:v1:old:c (set id-post:c)]`init)
+      [%x %v2 %channels full=?(~ [%full ~])]
+    ``channels-2+!>(`channels:v1:old:c`(uv-channels:utils v-channels ?=(^ full.pole)))
     ::
-        [%x %v4 %init ~]
-      =/  init  [(uv-channels-3:utils v-channels |) hidden-posts]
-      ``noun+!>(`[channels:c (set id-post:c)]`init)
+      [%x %v3 %channels full=?(~ [%full ~])]
+    ``channels-3+!>(`channels:c`(uv-channels-2:utils v-channels ?=(^ full.pole)))
+    ::
+    ::  /x/v/init: get unreads and unversioned channels
+    ::
+      [%x ?(%v0 %v1) %init ~]
+    ``noun+!>([unreads (uv-channels-1:utils v-channels)])
+    ::
+      [%x %v2 %init ~]
+    ``noun+!>([unreads (uv-channels:utils v-channels |)])
+    ::
+      [%x %v3 %init ~]
+    =/  init  [(uv-channels:utils v-channels |) hidden-posts]
+    ``noun+!>(`[channels:v1:old:c (set id-post:c)]`init)
+    ::
+      [%x %v4 %init ~]
+    =/  init  [(uv-channels-2:utils v-channels |) hidden-posts]
+    ``noun+!>(`[channels:c (set id-post:c)]`init)
     ::
       [%x ?(%v0 %v1) %hidden-posts ~]  ``hidden-posts+!>(hidden-posts)
       [%x ?(%v0 %v1) %unreads ~]  ``channel-unreads+!>(unreads)
-      [%x v=?(%v0 %v1 %v2) =kind:c ship=@ name=@ rest=*]
+      [%x v=?(%v0 %v1 %v2 %v3) =kind:c ship=@ name=@ rest=*]
     =/  =ship  (slav %p ship.pole)
     (ca-peek:(ca-abed:ca-core kind.pole ship name.pole) rest.pole v.pole)
   ::
-      [%u ?(%v0 %v1 %v2) =kind:c ship=@ name=@ ~]
+      [%u ?(%v0 %v1 %v2 %v3) =kind:c ship=@ name=@ ~]
     =/  =ship  (slav %p ship.pole)
     ``loob+!>((~(has by v-channels) kind.pole ship name.pole))
-  ::
+    ::
+    ::  /x/v/heads: get the latest post in each channel
+    ::
       [%x %v2 %heads since=?(~ [u=@ ~])]
     =/  since=(unit id-post:c)
       ?~  since.pole  ~
       ?^  tim=(slaw %da u.since.pole)  `u.tim
       `(slav %ud u.since.pole)
     :^  ~  ~  %channel-heads
+    !>  ^-  channel-heads:v7:old:c
+    (murn ~(tap by v-channels) (cury channel-head-1:utils since))
+    ::
+      [%x %v3 %heads since=?(~ [u=@ ~])]
+    =/  since=(unit id-post:c)
+     ?~  since.pole  ~
+      ?^  tim=(slaw %da u.since.pole)  `u.tim
+      `(slav %ud u.since.pole)
```

Comment:
```suggestion
    =/  since=(unit id-post:c)
      ?~  since.pole  ~
      ?^  tim=(slaw %da u.since.pole)  `u.tim
      `(slav %ud u.since.pole)
```